[Bug]: Update unregister logic

### Current Behavior

- Currently if user is involved in any project, register logic returns SQL error

### Expected Behavior

- If user trying to unregister, check if user has participated in any projects. If so, then returns 400 error 
- If user doesn't participate in any projects, then delete saved user-repos first and then delete user.


### Steps To Reproduce

1. Add user as member of any project
2. Or create new project (owner)
3. Trying to unregister

### Relevant log output

```shell
raise exc_value.with_traceback(exc_tb)
  File "/usr/local/lib/python3.11/site-packages/sqlalchemy/orm/session.py", line 4449, in _flush
    flush_context.execute()
  File "/usr/local/lib/python3.11/site-packages/sqlalchemy/orm/unitofwork.py", line 466, in execute
    rec.execute(self)
  File "/usr/local/lib/python3.11/site-packages/sqlalchemy/orm/unitofwork.py", line 589, in execute
    self.dependency_processor.process_deletes(uow, states)
  File "/usr/local/lib/python3.11/site-packages/sqlalchemy/orm/dependency.py", line 567, in process_deletes
    self._synchronize(
  File "/usr/local/lib/python3.11/site-packages/sqlalchemy/orm/dependency.py", line 625, in _synchronize
    sync.clear(dest, self.mapper, self.prop.synchronize_pairs)
  File "/usr/local/lib/python3.11/site-packages/sqlalchemy/orm/sync.py", line 88, in clear
    raise AssertionError(
AssertionError: Dependency rule on column 'user.id' tried to blank-out primary key column 'project_user.user_id' on instance '<ProjectUser at 0xffff8d75dd90>'
```

### Anything else?

_No response_

### Issue Submission Checklist

- [x] I have assigned this issue to myself or the appropriate person.
- [x] I have added the appropriate labels for this issue.

--- a/src/auth/service.py
+++ b/src/auth/service.py
@@ -13,11 +13,13 @@ from auth.util.redis import (
     save_token_to_redis,
 )
 from src.config.config import ACCESS_TOKEN_EXPIRE_MINUTES, FRONTEND_URL, IS_LOCAL
+from src.project_user.repository import find_all_projects_by_user_id
 from src.response.error_definitions import (
     AccessTokenNotFound,
     GitHubAccessTokenError,
     GitHubCredentialCodeNotFound,
     InvalidRefreshToken,
+    ProjectUserExist,
     UserAlreadyExist,
     UserNotFound,
 )
@@ -28,6 +30,7 @@ from src.user.repository import (
 )
 from src.user.schemas import UserReq, UserRes
 from src.user.service import create_user, update_user
+from src.user_repository.repository import delete_all_repositories_by_user_id
 
 GITHUB_OAUTH_REDIS = "github_oauth"
 REFRESH_TOKEN_REDIS = "refresh_token"
@@ -187,7 +190,7 @@ async def login(db: Session, access_token: Optional[str] = Cookie(None)):
     await save_token_to_redis(REFRESH_TOKEN_REDIS, existing_user.id, refresh_token)
 
     user_res = UserRes.model_validate(existing_user)
-    
+
     return AuthRes(
         user=user_res, access_token=access_token, refresh_token=refresh_token
     )
@@ -263,6 +266,14 @@ async def unregister(user_id: int, db: Session):
     """
     Unregister user
     """
+    # Check if user has project
+    project_list = find_all_projects_by_user_id(db, user_id)
+    if project_list:
+        raise ProjectUserExist()
+
+    # Delete user repo
+    delete_all_repositories_by_user_id(db, user_id)
+
     # Delete existing refresh token stored in redis
     redis_refresh_token = await get_token_from_redis(REFRESH_TOKEN_REDIS, user_id)
     await delete_token_from_redis(redis_refresh_token)

--- a/src/response/error_definitions.py
+++ b/src/response/error_definitions.py
@@ -233,6 +233,14 @@ class IssueReschedulingAlreadyExist(ConflictException):
         )
 
 
+class ProjectUserExist(ConflictException):
+    def __init__(self):
+        super().__init__(
+            title="프로젝트 참여 중",
+            detail="탈퇴하기 전에 참여 중인 모든 프로젝트에서 나가야 합니다. 프로젝트 목록에서 나가기를 완료해주세요.",
+        )
+
+
 """
 415 UNSUPPORTED_MEDIA_TYPE
 """
@@ -318,6 +326,7 @@ class ParseJsonFromResponseError(InternalServerErrorException):
             title="JSON 파싱 오류",
             detail="서버에서 응답을 JSON 형식으로 파싱하는 중 문제가 발생했습니다. 서버 로그를 확인해 주세요.",
         )
+
 
 class RecommendAssigneeError(InternalServerErrorException):
     def __init__(self):

--- a/src/user_repository/repository.py
+++ b/src/user_repository/repository.py
@@ -49,3 +49,18 @@ def find_all_repositories_by_user_id(db: Session, user_id: int) -> UserRepositor
         logger.error(f"Database error: {e}")
         db.rollback()
         raise SQLError()
+
+
+def delete_all_repositories_by_user_id(db: Session, user_id: int) -> None:
+    try:
+        repos = db.query(UserRepository).filter(UserRepository.user_id == user_id).all()
+
+        if repos:
+            for repo in repos:
+                db.delete(repo)
+            db.commit()
+
+    except SQLAlchemyError as e:
+        logger.error(f"Database error: {e}")
+        db.rollback()
+        raise SQLError()

